From 6e19a238bc4743964029a8ff36b878d3295e7a28 Mon Sep 17 00:00:00 2001
From: Samuel Holland <samuel@sholland.org>
Date: Thu, 18 Jun 2020 18:24:06 -0500
Subject: [PATCH] drivers: allwinner: axp: Enable regulators more judiciously

Some regulators should not be enabled at boot, even if they have a
phandle, such as when they have timing or sequencing requirements.

For example, the Ethernet PHY on the Orange Pi 3 has two power supplies:
one from the PMIC and one from a gpio-controlled regulator. The
gpio-controlled regulator must be enabled for the PHY to initialize
properly. The code here breaks that.

Regulators should need to opt in to being enabled at boot.

Signed-off-by: Samuel Holland <samuel@sholland.org>
Change-Id: Iffd4f06b5d4806d4636d73f960c129140ba34bcb
---
 drivers/allwinner/axp/common.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/allwinner/axp/common.c b/drivers/allwinner/axp/common.c
index cbd2e88185..df69a76324 100644
--- a/drivers/allwinner/axp/common.c
+++ b/drivers/allwinner/axp/common.c
@@ -98,8 +98,6 @@ static int setup_regulator(const void *fdt, int node,
 
 static bool should_enable_regulator(const void *fdt, int node)
 {
-	if (fdt_getprop(fdt, node, "phandle", NULL) != NULL)
-		return true;
 	if (fdt_getprop(fdt, node, "regulator-always-on", NULL) != NULL)
 		return true;
 	return false;
